api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/widgets_controller.rb

```diff
@@ -63,10 +65,17 @@ def widgets
       @widgets ||= MnoEnterprise::Impac::Dashboard.find(params[:dashboard_id]).widgets
     end
 
-    def format_attrs(whitelist)
-      attrs = (params[:widget] || {}).select { |k,v| whitelist.include?(k.to_s) }
-      attrs['settings'] = widget ? widget.settings || {} : {}
-      attrs['settings'].merge!(attrs['metadata']) if attrs['metadata']
-      attrs.except!('metadata')
+    def widget_create_params
+      params.require(:widget).permit(*[:widget_category]).tap do |whitelisted|
+        whitelisted[:settings] = params[:widget][:metadata] || {}
+      end
+      .except(:metadata)
+    end
```

Comment:
:heart: much better with strong params :wink: 

Could `permit(:widget_category)` work? Or is this a neat trick?